tiledb/api/src/query_arrow/fields.rs

```diff
@@ -0,0 +1,171 @@
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use arrow::array as aa;
+use arrow::datatypes as adt;
+
+use super::QueryBuilder;
+
+/// Default field capacity is 10MiB
+const DEFAULT_CAPACITY: usize = 1024 * 1024 * 10;
```

Comment:
+1. I didn't spend any time thinking too hard on this, and as you noticed I removed the entire `weight` concept to avoid going down a rabbit hole. Currently, buffer allocation is basically hard coded and done inside the query/querybuilder interface. My general thought for these is that we'd move buffer allocation out of the query interface and into something like that query adaptor interface.